packages/frontend/src/fetcher/on-chain/index.ts

```diff
@@ -108,62 +112,70 @@ class Fetcher implements IPartialFetcher {
     }
 
     public async fetchAnswersHistory({
-        provider,
+        nodeClient,
         realityV3Address,
         questionId,
     }: FetchAnswersHistoryParams): Promise<RealityResponse[]> {
         if (!realityV3Address || !questionId) return [];
-        const { chainId } = await provider.getNetwork();
+        const chainId = await nodeClient.getChainId();
         enforce(
             chainId in SupportedChainId,
             `unsupported chain with id ${chainId}`
         );
-        const realityContract = new Contract(
-            realityV3Address,
-            REALITY_ETH_V3_ABI,
-            provider
-        );
 
-        let toBlock = await provider.getBlockNumber();
+        let toBlock = await nodeClient.getBlockNumber();
         let fromBlock = toBlock - LOGS_BLOCKS_SIZE;
         const answersFromLogs: RealityResponse[] = [];
         try {
             while (true) {
-                const logs = await provider.getLogs({
-                    address: realityContract.address,
+                // FIXME: check if this implementation still works properly
+                const newAnwersEventLogs = await nodeClient.getLogs({
```

Comment:
Pretty sure this implementation works properly even though we might want to reduce the event interval size on which the logs are searched as some node providers don't support large values.

Reply:
Set to `5000`.